src/main/java/com/exercises/third/core/DependencyGraph.java

```diff
@@ -0,0 +1,70 @@
+package com.exercises.third.core;
+
+import com.exercises.third.view.ViewOption;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+import java.util.List;
+import java.util.HashSet;
+import java.util.HashMap;
+import java.util.ArrayList;
+import java.util.Collections;
+
+import static com.exercises.third.utils.PrintUtils.ioException;
+import static com.exercises.third.utils.Utils.parseJson;
+import static com.exercises.third.utils.Utils.readBytesFromFile;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class DependencyGraph {
+    private static final String RELATIVE_PATH = "src/main/resources/";
+    private ViewOption viewOption;
+
+    public DependencyGraph(ViewOption viewOption) {
+        this.viewOption = viewOption;
+    }
+
+    public List<Node> buildGraph(String fileName) {
+        Map<String, List<String>> dependencyMap= new HashMap<>();
+
+        try {
+            var content = readBytesFromFile(RELATIVE_PATH + fileName);
+            dependencyMap = parseJson(content);
+        } catch (IOException e) {
+            ioException(e);
+        }
+
+        List<Node> roots = new ArrayList<>();
+
+        for (String pkg : dependencyMap.keySet()) {
```

Comment:
Don't use short names of variables. There are people with pure knowledge of English. There is additional cognitive volume for reading such kind of stuff.